crates/ty_python_semantic/src/types/instance.rs

```diff
@@ -14,7 +14,10 @@ pub(super) use synthesized_protocol::SynthesizedProtocolType;
 impl<'db> Type<'db> {
     pub(crate) fn instance(db: &'db dyn Db, class: ClassType<'db>) -> Self {
         if class.class_literal(db).0.is_protocol(db) {
-            Self::ProtocolInstance(ProtocolInstanceType(Protocol::FromClass(class)))
+            Self::ProtocolInstance(ProtocolInstanceType {
+                inner: Protocol::FromClass(class),
+                _phantom: PhantomData,
+            })
```

Comment:
maybe create `ProtocolInstanceType::from_class()` and `ProtocolInstanceType::synthesized()` constructor functions (with comments saying that they should be kept private to the module)? The `_phantom: PhantomData` thing at every callsite feels like it gets a bit repetitive and boilerplate-y

Reply:
Done. Did this for `NominalInstanceType` too